hibernate-core/src/main/java/org/hibernate/dialect/H2Dialect.java

```diff
@@ -118,7 +118,8 @@ public H2Dialect() {
 		registerColumnType( Types.DECIMAL, "decimal($p,$s)" );
 		registerColumnType( Types.NUMERIC, "decimal($p,$s)" );
 		registerColumnType( Types.DOUBLE, "double" );
-		registerColumnType( Types.FLOAT, "float" );
+		// H2 does not have a float type. Instead, it's mapping floats as alias to double
+		registerColumnType( Types.FLOAT, "double" );
```

Comment:
I would do things differently, and register h2 "real" database type to java FLOAT: 
`registerColumnType(Types.FLOAT, "real");`

FYI this what has been done for the [JHipster ](https://www.jhipster.tech) generator, and thus what is used by many generated applications: https://github.com/jhipster/jhipster/blob/master/jhipster-framework/src/main/java/io/github/jhipster/domain/util/FixedH2Dialect.java#L36